### PR TITLE
CR-1136632 : Get tiles for AIE_profile_settings.tile_based_interface_tile_metrics and remove stray white space characters in configurations

### PR DESCRIPTION
**Pull request by IshitaGhosh**
#### Problem solved by the commit
Tiles specified with AIE_profile_settings.tile_based_interface_tile_metrics were not selected for profiling.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR 1136632 reported the issue

#### How problem was solved, alternative solutions (if any) and why they were rejected
For SHIM profiling, the specified tiles are now considered and stream ID s are also set for those.
This PR solves another issue : if stray white-space characters are present in AIE_profile_settings metrics configurations, it caused the comparison with valid metrics to fail. This resulted in selection of default metrics even if an otherwise valid metric was specified.

#### What has been tested and how, request additional testing if necessary
Reported test with tile ranges and also particular tile specification.

### CODE REVIEW

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 1]
```diff
@@ -1196,87 +1195,157 @@ namespace xdp {
...
+          if ((mChannelId >= 0) && (mChannelId != streamId))
+            continue;
+
+          // Make sure it's desired polarity
+          // NOTE: input = slave (data flowing from PLIO)
+          //       output = master (data flowing to PLIO)
+          if ((isMaster && (metrics[i][2] == "input_bandwidths"))
+              || (isMaster && (metrics[i][2] == "input_stalls_idle"))
+              || (!isMaster && (metrics[i][2] == "output_bandwidths"))
+              || (!isMaster && (metrics[i][2] == "output_stalls_idle")))
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp:1254]: These are not the correct metric set names as the "*_idle" ones do not exist anymore.

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 2]
```diff
@@ -1196,87 +1195,157 @@ namespace xdp {
...
+          if ((mChannelId >= 0) && (mChannelId != streamId))
+            continue;
+
+          // Make sure it's desired polarity
+          // NOTE: input = slave (data flowing from PLIO)
+          //       output = master (data flowing to PLIO)
+          if ((isMaster && (metrics[i][1] == "input_bandwidths"))
+                || (isMaster && (metrics[i][1] == "input_stalls_idle"))
+                || (!isMaster && (metrics[i][1] == "output_bandwidths"))
+                || (!isMaster && (metrics[i][1] == "output_stalls_idle")))
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp:1323]: Ditto here. Remove the *_idle ones.

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 3]
```diff
@@ -1313,6 +1382,304 @@ namespace xdp {
...
+        if ((channelId >= 0) && (channelId != streamId))
+          continue;
+
+        // Make sure it's desired polarity
+        // NOTE: input = slave (data flowing from PLIO)
+        //       output = master (data flowing to PLIO)
+        if ((isMaster && (metrics[i][2] == "input_bandwidths"))
+            || (isMaster && (metrics[i][2] == "input_stalls_idle"))
+            || (!isMaster && (metrics[i][2] == "output_bandwidths"))
+            || (!isMaster && (metrics[i][2] == "output_stalls_idle")))
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp:1549]: Remove *_idle ones

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp [hunk 4]
```diff
@@ -1313,6 +1382,304 @@ namespace xdp {
...
+          if ((channelId >= 0) && (channelId != streamId))
+            continue;
+
+          // Make sure it's desired polarity
+          // NOTE: input = slave (data flowing from PLIO)
+          //       output = master (data flowing to PLIO)
+          if ((isMaster && (metrics[i][1] == "input_bandwidths"))
+              || (isMaster && (metrics[i][1] == "input_stalls_idle"))
+              || (!isMaster && (metrics[i][1] == "output_bandwidths"))
+              || (!isMaster && (metrics[i][1] == "output_stalls_idle")))
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp:1635]: Same here. Why is this in four different places in the code?!?

**Review by pgschuey — CHANGES_REQUESTED** [review]

**Review by jvillarre — APPROVED** [review]
Approving so testing on the switches can proceed, but on the condition that additional clean up will come in future pull requests when testing feedback is incorporated.

File: src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp (hunk 1 above)
Comment (IshitaGhosh) [reply src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp]: I have removed the comparison with *_idle metrics.

**Pull request merged**